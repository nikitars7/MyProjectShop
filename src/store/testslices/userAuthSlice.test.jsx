import authSliceReducer, { login, logOut } from "../slices/userAuthSlice";
describe("filterSlice", () => {
  test("should return default state", () => {
    const result = authSliceReducer(undefined, { type: "" });
    expect(result).toEqual({ email: null, token: null, id: null });
    expect(result).toMatchSnapshot();
  });
  test("should return authorized state", () => {
    const action = {
      type: login.type,
      payload: {
        email: "som@gmail.com",
        token: "123456789",
        id: "1",
      },
    };
    const result = authSliceReducer(
      { email: null, token: null, id: null },
      action
    );
    expect(result).toEqual({
      email: "som@gmail.com",
      token: "123456789",
      id: "1",
    });
    expect(result).toMatchSnapshot();
  });
  test("should logout", () => {
    const action = {
      type: logOut.type,
      payload: {
        email: null,
        token: null,
        id: null,
      },
    };
    const result = authSliceReducer(
      {
        email: "som@gmail.com",
        token: "123456789",
        id: "1",
      },
      action
    );
    expect(result).toEqual({
      email: null,
      token: null,
      id: null,
    });
    expect(result).toMatchSnapshot();
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
});
