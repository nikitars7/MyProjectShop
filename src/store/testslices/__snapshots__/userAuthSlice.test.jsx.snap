// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filterSlice should logout 1`] = `
Object {
  "email": null,
  "id": null,
  "token": null,
}
`;

exports[`filterSlice should return authorized state 1`] = `
Object {
  "email": "som@gmail.com",
  "id": "1",
  "token": "123456789",
}
`;

exports[`filterSlice should return default state 1`] = `
Object {
  "email": null,
  "id": null,
  "token": null,
}
`;
